engine/src/test/java/com/redhat/cloud/notifications/processors/email/EmailAggregatorTest.java

```diff
@@ -207,23 +243,35 @@ private void verifyRecipientsResolverInteractions(int legacyRecipientResolverInv
         }
     }
 
-    private Map<User, Map<String, Object>> aggregate() {
+    private Map<User, Map<String, Object>> aggregate(boolean useEventAggregationCriteria) {
         Map<User, Map<String, Object>> result = new HashMap<>();
         emailAggregationRepository.addEmailAggregation(TestHelpers.createEmailAggregation("org-1", "rhel", "policies", RandomStringUtils.random(10), RandomStringUtils.random(10)));
         emailAggregationRepository.addEmailAggregation(TestHelpers.createEmailAggregation("org-1", "rhel", "policies", RandomStringUtils.random(10), RandomStringUtils.random(10)));
         emailAggregationRepository.addEmailAggregation(TestHelpers.createEmailAggregation("org-1", "rhel", "policies", RandomStringUtils.random(10), RandomStringUtils.random(10)));
         emailAggregationRepository.addEmailAggregation(TestHelpers.createEmailAggregation("org-1", "rhel", "policies", RandomStringUtils.random(10), RandomStringUtils.random(10)));
 
         emailAggregationRepository.addEmailAggregation(TestHelpers.createEmailAggregation("org-2", "rhel", "policies", RandomStringUtils.random(10), RandomStringUtils.random(10)));
+
+        for (int i = 0; i < 4; i++) {
+            JsonObject payload = TestHelpers.createEmailAggregation("org-1", "rhel", "policies", RandomStringUtils.random(10), RandomStringUtils.random(10)).getPayload();
+            resourceHelpers.addEventEmailAggregation("org-1", "rhel", "policies", payload, false);
+        }
+        JsonObject payload = TestHelpers.createEmailAggregation("org-2", "rhel", "policies", RandomStringUtils.random(10), RandomStringUtils.random(10)).getPayload();
+        resourceHelpers.addEventEmailAggregation("org-2", "rhel", "policies", payload, false);
+
+        EmailAggregationKey aggregationKey = AGGREGATION_KEY;
+        if (useEventAggregationCriteria) {
+            Application policiesApp = resourceHelpers.findApp("rhel", "policies");
+            aggregationKey = new EventAggregationCriteria(AGGREGATION_KEY.getOrgId(), policiesApp.getBundleId(), policiesApp.getId(), AGGREGATION_KEY.getBundle(), AGGREGATION_KEY.getApplication());
+        }
         result.putAll(emailAggregator.getAggregated(application.getId(), aggregationKey, DAILY, LocalDateTime.now(ZoneOffset.UTC).minusMinutes(1), LocalDateTime.now(ZoneOffset.UTC).plusMinutes(1)));
         return result;
     }
 
-    @CacheInvalidate(cacheName = "recipients-resolver-results")
```

Comment:
Unless something changed in quarkus-cache, the old code should be synchronous because the method returns `void` and not `Uni<Void>`. Anyway, the new code is fine so please don't change anything!